aws/provider.go

```diff
@@ -835,6 +835,7 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		config.KinesisEndpoint = endpoints["kinesis"].(string)
 		config.KmsEndpoint = endpoints["kms"].(string)
 		config.LambdaEndpoint = endpoints["lambda"].(string)
+		config.PricingEndpoint = endpoints["pricing"].(string)
```

Comment:
It doesn't look like everything is setup to correctly support the custom endpoint here. There are a few additional things required as shown in #4716. To not setup the customized endpoint, you can simply add handling similar to #4049